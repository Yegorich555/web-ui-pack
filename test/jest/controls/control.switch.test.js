import { WUPSwitchControl } from "web-ui-pack";
import { initTestBaseControl } from "./baseControlTest";
import testSwitchControl from "./control.switchTest";

/** @type WUPSwitchControl */
let testEl;
initTestBaseControl({ type: WUPSwitchControl, htmlTag: "wup-switch", onInit: (e) => (testEl = e) });

describe("control.switch", () => {
  testSwitchControl(() => testEl);
});
