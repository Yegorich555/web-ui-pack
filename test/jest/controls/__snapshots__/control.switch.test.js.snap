// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`control.switch $ariaDetails 1`] = `
<wup-switch>
  <label
    for="txt1"
  >
    <input
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
    <span
      class="wup-hidden"
    >
      Some details for screen-readers
    </span>
  </label>
</wup-switch>
`;

exports[`control.switch $ariaDetails 2`] = `
<wup-switch>
  <label
    for="txt1"
  >
    <input
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
</wup-switch>
`;

exports[`control.switch $ariaSpeak 1`] = `
<wup-switch>
  <label
    for="txt1"
  >
    <input
      aria-describedby="my-test-id txt2 my-test-id2"
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
  <section
    aria-atomic="true"
    aria-live="off"
    class="wup-hidden"
    id="txt2"
  >
    Hello
  </section>
</wup-switch>
`;

exports[`control.switch $ariaSpeak 2`] = `
<wup-switch>
  <label
    for="txt1"
  >
    <input
      aria-describedby="my-test-id my-test-id2"
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
</wup-switch>
`;

exports[`control.switch $show/$hide error 1`] = `
<wup-switch>
  <label
    for="txt1"
  >
    <input
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
</wup-switch>
`;

exports[`control.switch $show/$hide error 2`] = `
<wup-switch
  invalid=""
>
  <label
    for="txt1"
  >
    <input
      aria-describedby="txt2"
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
  <wup-popup
    aria-atomic="true"
    aria-live="off"
    error=""
    id="txt2"
    role="alert"
  >
    <span
      class="wup-hidden"
    />
    <span>
      Some custom message here
    </span>
  </wup-popup>
</wup-switch>
`;

exports[`control.switch $show/$hide error 3`] = `
<wup-switch>
  <label
    for="txt1"
  >
    <input
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
  <wup-popup
    aria-atomic="true"
    aria-live="off"
    error=""
    id="txt2"
    open=""
    role="alert"
    show=""
    style="display: none;"
  >
    <span
      class="wup-hidden"
    />
    <span>
      Some custom message here
    </span>
  </wup-popup>
</wup-switch>
`;

exports[`control.switch common tests render + styles 1`] = `<wup-switch />`;

exports[`control.switch common tests render + styles 2`] = `
<style>
  .wup-hidden, [wup-hidden] {
position: absolute;
height:1px; width:1px;
overflow:hidden;
clip:rect(1px,1px,1px,1px);
min-width:initial;
padding:0;}

  
[wup-icon] {
  display: inline-block;
  cursor: pointer;
  box-shadow: none;
  border: none;
  margin: 0;
  padding: 0;
  width: var(--icon-hover-r, 2em);
  height: var(--icon-hover-r, 2em);
  background: none;
  outline: none;
  border-radius: 50%;
}
[wup-icon]:after {
  content: "";
  display: inline-block;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  background: var(--icon, #000);
  -webkit-mask-size: var(--icon-size, 1em);
  mask-size: var(--icon-size, 1em);
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;

  -webkit-mask-image: var(--icon-img);
  mask-image: var(--icon-img);
}
[wup-icon]:focus {
   box-shadow: inset 0 0 0 99999px var(--icon-focus-bg);
}
[wup-icon]:focus:after {
   background: var(--icon-hover, var(--icon, #000));
}
@media (hover: hover) and (pointer: fine) {
  [wup-icon]:hover {
   box-shadow: inset 0 0 0 99999px var(--icon-hover-bg);
  }
  [wup-icon]:hover:after {
    background: var(--icon-hover, var(--icon, #000));
  }
  [wup-icon]:focus:hover {
   opacity: 0.9;
  }
}

  :root {
          --base-focus: #00778d;
          --base-btn-bg: #009fbc;
          --base-btn-text: #fff;
          --base-btn-focus: #005766;
          --base-btn2-bg: #6c757d;
          --base-btn2-text: #fff;
          --base-btn3-bg: none;
          --base-btn3-text: inherit;
          --base-sep: #e4e4e4;
          --base-margin: 20px;
          --border-radius: 6px;
          --anim-t: 200ms;
          --anim: var(--anim-t) cubic-bezier(0, 0, 0.2, 1) 0ms;
          --icon-hover-r: 30px;
          --icon-hover-bg: #0001;
          --icon-focus-bg: #0000001a;
          --icon-size: 14px;
          --menu-hover-text: inherit;
          --menu-hover-bg: #f1f1f1;
        }
        [wupdark] {
          --base-btn-focus: #bdbdbd;
          --base-sep: #141414;
          --icon: #fff;
          --icon-hover-bg: #fff1;
          --icon-focus-bg: #fff2;
          --scroll: #fff2;
          --scroll-hover: #fff3;
          --menu-hover-text: inherit;
          --menu-hover-bg: #222a36;
        }
  :root {
        --ctrl-padding: 1.5em 1em 0.5em 1em;
        --ctrl-focus: var(--base-focus);
        --ctrl-focus-label: var(--ctrl-focus);
        --ctrl-selected: var(--ctrl-focus);
        --ctrl-label: #5e5e5e;
        --ctrl-icon: var(--ctrl-label);
        --ctrl-icon-size: 1em;
        --ctrl-text: inherit;
        --ctrl-bg: #fff;
        --ctrl-border: #e6e6e6;
        --ctrl-border-radius: var(--border-radius);
        --ctrl-err: #ad0000;
        --ctrl-err-bg: #fff4fa;
        --ctrl-err-valid: green;
        --ctrl-invalid-border: red;
        --wup-icon-cross: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='768' height='768'%3E%3Cpath d='M674.515 93.949a45.925 45.925 0 0 0-65.022 0L384.001 318.981 158.509 93.487a45.928 45.928 0 0 0-65.022 0c-17.984 17.984-17.984 47.034 0 65.018l225.492 225.494L93.487 609.491c-17.984 17.984-17.984 47.034 0 65.018s47.034 17.984 65.018 0l225.492-225.492 225.492 225.492c17.984 17.984 47.034 17.984 65.018 0s17.984-47.034 0-65.018L449.015 383.999l225.492-225.494c17.521-17.521 17.521-47.034 0-64.559z'/%3E%3C/svg%3E");
        --wup-icon-check: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='768' height='768'%3E%3Cpath d='M37.691 450.599 224.76 635.864c21.528 21.32 56.11 21.425 77.478 0l428.035-426.23c21.47-21.38 21.425-56.11 0-77.478s-56.11-21.425-77.478 0L263.5 519.647 115.168 373.12c-21.555-21.293-56.108-21.425-77.478 0s-21.425 56.108 0 77.478z'/%3E%3C/svg%3E");
        --wup-icon-dot: url("data:image/svg+xml,%3Csvg viewBox='0 0 100 100' xmlns='http://www.w3.org/2000/svg'%3E%3Ccircle cx='50' cy='50' r='20'/%3E%3C/svg%3E");
        --wup-icon-back: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='768' height='768'%3E%3Cpath d='m509.8 16.068-329.14 329.14c-21.449 21.449-21.449 56.174 0 77.567l329.14 329.14c21.449 21.449 56.174 21.449 77.567 0s21.449-56.174 0-77.567l-290.36-290.36 290.36-290.36c21.449-21.449 21.449-56.173 0-77.567-21.449-21.394-56.173-21.449-77.567 0z'/%3E%3C/svg%3E");
      }
      [wupdark] {
        --ctrl-bg: none;
        --ctrl-border: #104652;
        --ctrl-label: #919191;
        --ctrl-icon: var(--ctrl-label);
        --ctrl-focus-label: #25a1b6;
        --ctrl-selected: #25a1b6;
        --ctrl-err: #ff6666;
        --ctrl-err-bg: #000;
        --ctrl-err-valid: #54d754;
      }
  :root {
      --ctrl-switch-padding: 1em;
      --ctrl-switch-on: #fff;
      --ctrl-switch-off: #fff;
      --ctrl-switch-off-bg: #9f9f9f;
      --ctrl-switch-on-bg: var(--ctrl-focus);
      --ctrl-switch-shadow: #0003;
      --ctrl-switch-h: var(--ctrl-icon-size);
      --ctrl-switch-w: calc(var(--ctrl-icon-size) * 2.8);
      --ctrl-switch-r: calc(var(--ctrl-icon-size) * 1.4);
     }
    [wupdark] {
      --ctrl-switch-on: #e7e7e7;
      --ctrl-switch-off: #e7e7e7;
      --ctrl-switch-off-bg: #707070;
      --ctrl-switch-shadow: #000;
    }
  
      WUP-SWITCH {
        --ctrl-focus-border: var(--ctrl-focus);
        contain: style;
        display: block;
        flex: 1;
        margin: var(--base-margin) 0;
        box-shadow: 0 0 0 1px var(--ctrl-border);
        border-radius: var(--ctrl-border-radius);
        color: var(--ctrl-text);
        background: var(--ctrl-bg);
        cursor: pointer;
        -webkit-tap-highlight-color: transparent;
      }
      WUP-SWITCH strong,
      WUP-SWITCH legend {
        color: var(--ctrl-label);
        pointer-events: none;
      }
      WUP-SWITCH[invalid],
      WUP-SWITCH[invalid] &gt; [menu] {
        --ctrl-focus-border: var( --ctrl-invalid-border);
        box-shadow: 0 0 3px 0 var(--ctrl-focus-border);
      }
      WUP-SWITCH:focus-within,
      WUP-SWITCH:focus-within &gt; [menu] {
        box-shadow: 0 0 2px 1px var(--ctrl-focus-border);
      }
      WUP-SWITCH:focus-within strong,
      WUP-SWITCH:focus-within legend {
        color: var(--ctrl-focus-label);
        pointer-events: initial;
      }
      [disabled] WUP-SWITCH,
      WUP-SWITCH[disabled] {
        opacity: 0.6;
        cursor: not-allowed;
        -webkit-user-select: none;
        user-select: none;
      }
      [disabled] WUP-SWITCH &gt; label,
      WUP-SWITCH[disabled] &gt; label {
        pointer-events: none;
      }
      WUP-SWITCH label {
        display: flex;
        align-items: center;
        box-sizing: border-box;
        font: inherit;
        cursor: inherit;
        padding: var(--ctrl-padding);
        padding-top: 0;
        padding-bottom: 0;
      }
      WUP-SWITCH input,
      WUP-SWITCH textarea {
        padding: 0;
        margin: 0;
        cursor: inherit;
      }
      WUP-SWITCH [contenteditable=true] {
        margin: var(--ctrl-padding);
        margin-left: 0;
        margin-right: 0;
        cursor: inherit;
      }
      WUP-SWITCH strong {
        cursor: inherit;
      }
      WUP-SWITCH strong:empty {
        display: none;
      }
      WUP-SWITCH [aria-required=true] + strong:after,
      WUP-SWITCH fieldset[aria-required=true] &gt; legend:after {
        content: "*";
        font-size: larger;
        font-weight: bolder;
        line-height: 0;
      }
      WUP-SWITCH [error] {
        cursor: pointer;
        font-size: small;
        color: var(--ctrl-err);
        background: var(--ctrl-err-bg);
        overflow: auto;
        overflow: overlay;
        // text-shadow: 0 0 0 var(--ctrl-err);
      }
      WUP-SWITCH [error] ul {
        margin:0; padding:2px 4px 2px;
      }
      WUP-SWITCH [error] li {
        display: flex;
        align-items: center;
        margin-left: -5px;
      }
      WUP-SWITCH [error] li:before {
        content: '';
        --ctrl-icon-img: var(--wup-icon-dot);
        --ctrl-icon: var(--ctrl-err);
        
display: inline-block;
width: var(--ctrl-icon-size);
min-height: var(--ctrl-icon-size);
box-sizing: content-box;
margin: 0;
padding: 5px;
flex: 0 0 auto;
align-self: stretch;
border: none;
box-shadow: none;
background: var(--ctrl-icon);
-webkit-mask-size: var(--ctrl-icon-size);
mask-size: var(--ctrl-icon-size);
-webkit-mask-repeat: no-repeat;
mask-repeat: no-repeat;
-webkit-mask-position: center;
mask-position: center;
-webkit-mask-image: var(--ctrl-icon-img);
mask-image: var(--ctrl-icon-img);
      }
      WUP-SWITCH [error] li[valid] {
        color: var(--ctrl-err-valid);
      }
      WUP-SWITCH [error] li[valid]:before {
        content: '';
        --ctrl-icon-img: var(--wup-icon-check);
        --ctrl-icon: var(--ctrl-err-valid);
      }
      WUP-SWITCH:focus-within [error] {
        max-height: none;
      }
      @media (hover: hover) and (pointer: fine) {
        WUP-SWITCH:hover strong,
        WUP-SWITCH[hovered] strong{
          color: var(--ctrl-focus-label);
        }
        WUP-SWITCH:hover,
        WUP-SWITCH:hover&gt;[menu],
        WUP-SWITCH[hovered],
        WUP-SWITCH[hovered]&gt;[menu] {
          box-shadow: 0 0 2px 1px var(--ctrl-focus-border);
        }
      }
      @media not all and (prefers-reduced-motion) {
        WUP-SWITCH {
          transition: box-shadow var(--anim);
        }
      }
      WUP-SWITCH {
        background: none;
        cursor: pointer;
      }
      WUP-SWITCH[readonly] {
        cursor: initial;
      }
      WUP-SWITCH label {
        display: flex;
        gap: 0.5em;
        padding: var(--ctrl-switch-padding);
      }
      WUP-SWITCH strong {
        box-sizing: border-box;
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
        font-weight: normal;
        text-decoration: none;
        color: inherit;
      }
      WUP-SWITCH [bar] {
        display: inline-flex;
        align-items: center;
        overflow: visible;
        width: var(--ctrl-switch-w);
        min-width: var(--ctrl-switch-w);
        height: var(--ctrl-switch-h);
        border-radius: 999px;
        color: whitesmoke;
        background: var(--ctrl-switch-off-bg);
      }
      
WUP-SWITCH [thumb] {
    transform-style: preserve-3d;
  }
WUP-SWITCH [thumb]:before {
  z-index: -1;
  position: absolute;
  transform: translate(-50%, -50%) translateZ(-1px);
  left: 50%; top: 50%;
  width: var(--icon-hover-r, 2.4em);
  height: var(--icon-hover-r, 2.4em);
  background: var(--icon-hover, #0001);
  border-radius: 50%;
}
WUP-SWITCH:focus-within [thumb] {
  position: relative;
}
WUP-SWITCH:focus-within [thumb]:before {
  content: "";
}
@media (hover: hover) and (pointer: fine) {
  WUP-SWITCH:hover [thumb] {
    position: relative;
  }
  WUP-SWITCH:hover [thumb]:before {
    content: "";
    opacity: 0.8;
  }
}
      WUP-SWITCH [thumb] {
        z-index: 2;
        display: inline-block;
        height: var(--ctrl-switch-r);
        width: var(--ctrl-switch-r);
        background: var(--ctrl-switch-off);
        box-shadow: 0 1px 4px 0 var(--ctrl-switch-shadow);
        border-radius: 50%;
        transform: translateX(-1px);
      }
      WUP-SWITCH input { 
position: absolute;
height:1px; width:1px;
overflow:hidden;
clip:rect(1px,1px,1px,1px);
min-width:initial;
padding:0; }
      WUP-SWITCH[checked] [bar] {
        background-color: var(--ctrl-switch-on-bg);
      }
      WUP-SWITCH[checked] [thumb] {
        background: var(--ctrl-switch-on);
        transform: translateX(var(--ctrl-switch-w)) translateX(calc(-100% + 1px));
      }
      WUP-SWITCH[w-reverse] label {
        flex-direction: row-reverse;
      }
      WUP-SWITCH[w-reverse] strong {
        margin-right: auto;
      }
      @media not all and (prefers-reduced-motion) {
        WUP-SWITCH [bar] { transition: background-color var(--anim); }
        WUP-SWITCH [thumb] { transition: transform var(--anim); }
      }
</style>
`;

exports[`control.switch common tests snapshot of $defaults 1`] = `
{
  "autoComplete": null,
  "autoFocus": false,
  "clearActions": 2,
  "disabled": false,
  "focusDebounceMs": 100,
  "label": null,
  "name": null,
  "readOnly": false,
  "reverse": false,
  "storage": "local",
  "storageKey": "",
  "validateDebounceMs": 500,
  "validationCase": 13,
  "validationRules": {
    "$alwaysInvalid": [Function],
    "$alwaysValid": [Function],
    "required": [Function],
  },
  "validationShowAll": false,
  "validations": null,
}
`;

exports[`control.switch common tests snapshot of $defaults 2`] = `undefined`;

exports[`control.switch common tests snapshot of $defaults 3`] = `
[
  "defaultchecked",
  "disabled",
  "readonly",
  "w-autocomplete",
  "w-autofocus",
  "w-clearactions",
  "w-disabled",
  "w-focusdebouncems",
  "w-initvalue",
  "w-label",
  "w-name",
  "w-readonly",
  "w-reverse",
  "w-storage",
  "w-storagekey",
  "w-validatedebouncems",
  "w-validationcase",
  "w-validationrules",
  "w-validations",
  "w-validationshowall",
]
`;

exports[`control.switch validations options.validationCase 1`] = `
<wup-switch
  checked=""
>
  <label
    for="txt1"
  >
    <input
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong />
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
</wup-switch>
`;

exports[`control.switch validations options.validationCase 2`] = `
<wup-switch
  checked=""
  invalid=""
  required=""
>
  <label
    for="txt1"
  >
    <input
      aria-describedby="txt3"
      aria-required="true"
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong>
      First Input
    </strong>
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
  <wup-popup
    aria-atomic="true"
    aria-live="off"
    error=""
    id="txt3"
    open=""
    show=""
    style="display: none;"
  >
    <span
      class="wup-hidden"
    >
      Error for First Input:
    </span>
    <span>
      custom err here
    </span>
  </wup-popup>
</wup-switch>
`;

exports[`control.switch validations options.validationCase 3`] = `
<wup-switch
  invalid=""
>
  <label
    for="txt1"
  >
    <input
      aria-describedby="txt4"
      autocomplete="off"
      id="txt1"
      type="checkbox"
    />
    <strong>
      First Input
    </strong>
    <span
      bar=""
    >
      <span
        thumb=""
      />
    </span>
  </label>
  <wup-popup
    aria-atomic="true"
    aria-live="off"
    error=""
    id="txt4"
    open=""
    role="alert"
    show=""
    style="display: none;"
  >
    <span
      class="wup-hidden"
    >
      Error for First Input:
    </span>
    <span>
      Always error
    </span>
  </wup-popup>
</wup-switch>
`;

exports[`control.switch with form 1`] = `
<wup-form
  role="form"
>
  <wup-switch>
    <label
      for="txt1"
    >
      <input
        autocomplete="off"
        id="txt1"
        type="checkbox"
      />
      <strong>
        First Input
      </strong>
      <span
        bar=""
      >
        <span
          thumb=""
        />
      </span>
    </label>
  </wup-switch>
</wup-form>
`;
